hydra-node/exe/hydra-node/Main.hs

```diff
@@ -120,11 +121,11 @@ main = do
     let localhost = Host{hostname = show host, port}
      in withHeartbeat nodeId $ withOuroborosNetwork tracer localhost peers
 
-  withCardanoLedger ledgerConfig action = do
-    globals <-
-      newGlobals
-        <$> readJsonFileThrow shelleyGenesisFromJson (cardanoLedgerGenesisFile ledgerConfig)
-
+  withCardanoLedger ledgerConfig chainConfig action = do
+    let DirectChainConfig{networkId, nodeSocket} = chainConfig
+    -- NOTE: hydra-node will fail early if the conversion
```

Comment:
Better location for the comment, but pointless. You could add a comment on each IO action, that if it throws and not caught the process ends